app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,8 @@
+class LikesController < ApplicationController
+  def create
+    post = Post.find(params[:id])
+    like = current_user.likes.create(post: post)
+    like.save!
+    redirect_to user_post_url
+  end
```

Comment:
- [ ] Same here try to use the `if else` condition so that if it doesn't save, your app doesn't crash and you can give a custom message and redirect to a certain path.

```ruby
if like.save
 ...
else
 ...
end
```